cfme/utils/events.py

```diff
@@ -4,159 +4,68 @@
 
 """
 
-from cached_property import cached_property
-from contextlib import contextmanager
 from collections import Iterable
-from datetime import datetime
 from numbers import Number
-from sqlalchemy.sql.expression import func
 from time import sleep
 from threading import Thread, Event as ThreadEvent
 
 from cfme.utils.log import create_sublogger
+from manageiq_client.filters import Q
 
 logger = create_sublogger('events')
 
 
 class EventTool(object):
-    """EventTool serves as a wrapper to getting the events from the database.
+    """ EventTool serves as a wrapper to getting the events via REST API.
 
-    :var OBJECT_TABLE: Mapping of object types to tables and column names.
+    :var TARGET_TYPES: Mapping of object types to REST API collections.
     """
-    OBJECT_TABLE = {
-        # target_type: (table_name, name_column, id_column)
-        'VmOrTemplate': ('vms', 'name', 'id'),
-        'Host': ('hosts', 'name', 'id'),
-        'Service': ('services', 'name', 'id'),
+    TARGET_TYPES = {
+        # target_type: target_rest
+        'VmOrTemplate': 'vms',
+        'Host': 'hosts',
+        'Service': 'services',
     }
 
     def __init__(self, appliance):
         self.appliance = appliance
 
-    @property
-    def miq_event_definitions(self):
-        """``miq_event_definitions`` table."""
-        return self.appliance.db.client['miq_event_definitions']
-
     @property
     def event_streams(self):
-        """``event_streams`` table."""
-        return self.appliance.db.client['event_streams']
-
-    @cached_property
-    def event_streams_attributes(self):
-        """``event_streams`` columns and python's column types"""
-        self.appliance.db.client._table('event_streams')
-        event_table = [tbl for tbl in self.appliance.db.client.metadata.sorted_tables
-                       if tbl.name == 'event_streams'][-1]
-        return [(cl.name, cl.type.python_type) for cl in event_table.c.values()]
-
-    def query(self, *args, **kwargs):
-        """Wrapper for the SQLAlchemy query method."""
-        return self.appliance.db.client.session.query(*args, **kwargs)
-
-    @cached_property
-    def all_event_types(self):
-        """Returns a list of all possible events that can be used.
-
-        Returns:
-            A :py:class:`list` of :py:class:`str`.
-        """
-        return {q[0] for q in self.query(self.miq_event_definitions.name)}
+        """ Event streams REST API collection."""
+        return self.appliance.rest_api.collections.event_streams
```

Comment:
I doubt this is necessary. Event streams are available thru rest now.

Reply:
This just shortens the code.